cmd/agent/app/run.go

```diff
@@ -299,6 +299,9 @@ func StartAgent() error {
 	// Detect Cloud Provider
 	go util.DetectCloudProvider()
 
+	// Append version and timestamp to version history log file if this Agent is different than the last run version
+	go util.LogVersionHistory()
```

Comment:
why do a goroutine rather than direct function call?  The LogVersionHistory() shouldn't take long

Reply:
Do we prefer to use async whenever feasible or only when the function is going to take a while to finish?

This function doesn't run for a long time, but I thought making it run async can unblock the main thread.

Reply:
Change the routine to the regular sync call.

Reply:
IMO it would make sense to make this a goroutine, because this is a blocking IO operation.

Reply:
Probably the IO is fast enough here to not care either way. Both options seem equally fine to me.

Reply:
I originally asked for it to be made synchronous.  It's a small write, and we have no synchronization anywhere in our agent.  What happens if the agent is stopped while that routine is running?  We have no facility to wait until it's complete.
This problem is not limited to the originally proposed change, but I would advocate we not make it worse.

Reply:
Agree with Derek, won't change.